pavelib/quality.py

```diff
@@ -398,6 +516,64 @@ def _get_count_from_last_line(filename, file_type):
         return None
 
 
+def _get_safelint_counts(filename):
+    """
+    This returns a dict of violations from the safelint report.
+
+    Arguments:
+        filename: The name of the safelint report.
+
+    Returns:
+        A dict containing the following:
+            rules: A dict containing the count for each rule as follows:
+                violation-rule-id: N, where N is the number of violations
+            total: M, where M is the number of total violations
+
+    """
+    report_contents = _get_report_contents(filename)
+    rule_count_regex = re.compile(r"^(?P<rule_id>[a-z-]+):\s+(?P<count>\d+) violations", re.MULTILINE)
+    total_count_regex = re.compile(r"^(?P<count>\d+) violations total", re.MULTILINE)
+    violations = {'rules': {}}
+    for violation_match in rule_count_regex.finditer(report_contents):
+        try:
+            violations['rules'][violation_match.group('rule_id')] = int(violation_match.group('count'))
+        except ValueError:
+            violations['rules'][violation_match.group('rule_id')] = None
+    try:
+        violations['total'] = int(total_count_regex.search(report_contents).group('count'))
+    # An AttributeError will occur if the regex finds no matches.
+    # A ValueError will occur if the returned regex cannot be cast as a float.
+    except (AttributeError, ValueError):
+        violations['total'] = None
+    return violations
+
+
+def _get_safecommit_count(filename):
+    """
+    Returns the violation count from the safecommit report.
+
+    Arguments:
+        filename: The name of the safecommit report.
+
+    Returns:
+        The count of safecommit violations, or None if there is  a problem.
+
+    """
+    report_contents = _get_report_contents(filename)
+    validation_count = None
+    file_count_regex = re.compile(r"^(?P<count>\d+) violations total", re.MULTILINE)
+    try:
+        for count_match in file_count_regex.finditer(report_contents):
+            if validation_count is None:
+                validation_count = 0
+            validation_count += int(count_match.group('count'))
+    except ValueError:
+        validation_count = None
+    if validation_count is None and 'No files linted' in report_contents:
```

Comment:
This would be better put up before the regex and try block. That is, immediately return 0 if it sees the "No files linted" message in report_contents.